File: backend/src/main/java/com/example/nd/controller/AuthController.java
package com.example.nd.controller;

import com.example.nd.dto.*;
import com.example.nd.mapper.UserMapper;
import com.example.nd.model.User;
import com.example.nd.service.AuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
@Tag(name = "认证管理", description = "用户认证相关接口")
public class AuthController {

    @Autowired
    private AuthService authService;

    @PostMapping("/register")
    @Operation(summary = "用户注册", description = "用户名密码邮箱手机号注册")
    public ApiResponse<LoginResponse> register(@RequestBody RegisterRequest request) {
        LoginResponse response = authService.register(request);
        return ApiResponse.success(response);
    }

    @PostMapping("/login")
    @Operation(summary = "用户登录", description = "用户名密码登录")
    public ApiResponse<LoginResponse> login(@RequestBody LoginRequest request) {
        LoginResponse response = authService.login(request);
        return ApiResponse.success(response);
    }

    @PostMapping("/refresh")
    @Operation(summary = "令牌刷新", description = "使用刷新令牌获取新的访问令牌")
    public ApiResponse<RefreshTokenResponse> refreshToken(@RequestBody RefreshTokenRequest request) {
        RefreshTokenResponse response = authService.refreshToken(request);
        return ApiResponse.success(response);
    }

    @PostMapping("/logout")
    @Operation(summary = "用户登出", description = "用户登出")
    public ApiResponse<String> logout(@RequestBody RefreshTokenRequest request) {
        authService.logout(request.getRefreshToken());
        return ApiResponse.success("Logout successful");
    }

    @PostMapping("/forgot-password")
    @Operation(summary = "密码找回", description = "发送密码找回链接")
    public ApiResponse<String> forgotPassword(@RequestBody ForgotPasswordRequest request) {
        authService.forgotPassword(request);
        return ApiResponse.success("Reset link sent to your email");
    }

    @PostMapping("/reset-password")
    @Operation(summary = "密码重置", description = "重置密码")
    public ApiResponse<String> resetPassword(@RequestBody ResetPasswordRequest request) {
        authService.resetPassword(request);
        return ApiResponse.success("Password reset successful");
    }

    @PostMapping("/delete-account")
    @Operation(summary = "账号注销", description = "注销用户账号")
    public ApiResponse<String> deleteAccount(@RequestBody DeleteAccountRequest deleteRequest) {
        // 从Sa-Token获取当前登录用户ID
        Long userId = cn.dev33.satoken.stp.StpUtil.getLoginIdAsLong();
        authService.deleteAccount(userId, deleteRequest);
        return ApiResponse.success("Account deleted successfully");
    }
}

File: backend_python/app/controllers/auth_controller.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from app.database import get_db
from app.schemas import (
    RegisterRequest,
    LoginRequest,
    LoginResponse,
    RefreshTokenRequest,
    RefreshTokenResponse,
    ForgotPasswordRequest,
    ResetPasswordRequest,
    DeleteAccountRequest,
    ApiResponse,
    UserProfileResponse,
    UserUpdateRequest,
    ChangePasswordRequest,
)
from app.services.auth_service import AuthService
from app.services.user_service import UserService
from app.dependencies import get_current_user, get_current_active_user
from app.models import User

router = APIRouter(prefix="/auth", tags=["认证管理"])


@router.post("/register", response_model=dict)
def register(request: RegisterRequest, db: Session = Depends(get_db)):
    """用户注册"""
    auth_service = AuthService(db)
    try:
        result = auth_service.register(request)
        return {"code": 200, "message": "success", "data": result}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Registration failed: {str(e)}",
        )


@router.post("/login", response_model=dict)
def login(request: LoginRequest, db: Session = Depends(get_db)):
    """用户登录"""
    auth_service = AuthService(db)
    try:
        result = auth_service.login(request)
        return {"code": 200, "message": "success", "data": result}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Login failed: {str(e)}",
        )


@router.post("/refresh", response_model=dict)
def refresh_token(request: RefreshTokenRequest, db: Session = Depends(get_db)):
    """令牌刷新"""
    auth_service = AuthService(db)
    try:
        result = auth_service.refresh_token(request)
        return {"code": 200, "message": "success", "data": result}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Token refresh failed: {str(e)}",
        )


@router.post("/logout", response_model=dict)
def logout(request: RefreshTokenRequest, db: Session = Depends(get_db)):
    """用户登出"""
    auth_service = AuthService(db)
    try:
        auth_service.logout(request.refresh_token)
        return {"code": 200, "message": "success", "data": "Logout successful"}
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Logout failed: {str(e)}",
        )


@router.post("/forgot-password", response_model=dict)
def forgot_password(request: ForgotPasswordRequest, db: Session = Depends(get_db)):
    """密码找回"""
    auth_service = AuthService(db)
    try:
        auth_service.forgot_password(request)
        return {
            "code": 200,
            "message": "success",
            "data": "Reset link sent to your email",
        }
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Failed to send reset link: {str(e)}",
        )


@router.post("/reset-password", response_model=dict)
def reset_password(request: ResetPasswordRequest, db: Session = Depends(get_db)):
    """密码重置"""
    auth_service = AuthService(db)
    try:
        auth_service.reset_password(request)
        return {"code": 200, "message": "success", "data": "Password reset successful"}
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Password reset failed: {str(e)}",
        )


@router.post("/delete-account", response_model=dict)
def delete_account(
    request: DeleteAccountRequest,
    current_user: User = Depends(get_current_active_user),
    db: Session = Depends(get_db),
):
    """账号注销"""
    auth_service = AuthService(db)
    try:
        auth_service.delete_account(current_user.id, request)
        return {
            "code": 200,
            "message": "success",
            "data": "Account deleted successfully",
        }
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=f"Account deletion failed: {str(e)}",
        )
